### PR TITLE
Update Ruby 2.3.0 to 2.3.3 and test 2.4.0 on CI.

### PR DESCRIPTION
**Pull request by connorshea**
Happy holidays :)

### CODE REVIEW

**Comment by connorshea** [comment]
Apparently we have to wait for https://github.com/rails/rails/commit/47c49242913abbcdb8f798f53324769892d18c97 to get compatibility with Ruby 2.4/Rails 4.2.

File: .travis.yml [hunk 1]
```diff
@@ -5,6 +5,16 @@ before_install:
...
+  - 2.3.3
+  - 2.4.0
+gemfile:
+  - gemfiles/rack-2.gemfile
+  - gemfiles/rack-1.gemfile
+matrix:
+  exclude:
+    - rvm: 2.1
+      gemfile: gemfiles/rack-2.gemfile
+    - rvm: 2.4
```
Comment (bsedat) [inline .travis.yml]: Do we need to exclude "2.4.0" here?

Comment (connorshea) [inline .travis.yml]: ah, yeah my bad.

Comment (bsedat) [inline .travis.yml]: No worries! Thanks for helping keep this up to date. Happy new year!

**Pull request closed without merging**